sha2/tests/mod.rs

```diff
@@ -1,14 +1,61 @@
-use digest::dev::{feed_rand_16mib, fixed_reset_test};
-use digest::new_test;
+use digest::{
+    crypto_common::SerializableState,
+    dev::{feed_rand_16mib, fixed_reset_test},
+    new_test,
+};
 use hex_literal::hex;
 use sha2::{Digest, Sha224, Sha256, Sha384, Sha512, Sha512_224, Sha512_256};
 
+macro_rules! test_serializable_state {
+    ($name:ident, $hasher:ty) => {
+        #[test]
+        fn $name() {
+            let mut h = <$hasher>::new();
+
+            // First part of data.
+            feed_rand_16mib(&mut h);
```

Comment:
16 MiB is not a great value for testing serialization, we should use an odd number of bytes to check that buffer gets serialized properly. Also it's a bit too big in my opinion, we can simply use small fixed strings with length equal to block size plus several bytes.

Reply:
Test macro is moved to `traits`, input data is changed to `BlockSize + 1`. 
https://github.com/RustCrypto/traits/pull/1078/commits/793f9f268d220be0420e2c6b285a367ade048f88.